tests/workflow.spec.ts

```diff
@@ -51,4 +49,59 @@ test.describe(`Testfälle für den Test von workflows: Umgebung: ${process.env.U
       await Startseite.text_h2_Ueberschrift.click();
     })
   })  
+
+  test('SPSH-215 Passwort Reset', async ({ page}) => {
+    const Landing = new LandingPage(page);
+    const Login = new LoginPage(page);
+    const Startseite = new StartPage(page);
+    const UserManagement = new UserManagementPage(page);
+    const UserManagementDetail = new UserManagementDetailPage(page);
+    const Header = new HeaderPage(page);
+    const username_lastname = 'Max';
+    let new_password = '';
+
+    await test.step(`Portal öffnen ${URL_PORTAL}`, async () => {
+      await page.goto(URL_PORTAL);
+    })
+
+    await test.step(`Annmelden mit Administrator ${USER}`, async () => {
+      await Landing.button_Anmelden.click();
+      await Login.login(USER, PW); 
+      await expect(Startseite.text_h2_Ueberschrift).toBeVisible();
+    })
+
+    await test.step(`Benutzerverwaltung öffnen`, async () => {
+      await Startseite.card_item_schulportal_administration.click();
+    })
+
+    await test.step(`In der Benutzerverwaltung die Zeile für Benutzer ${username_lastname} anklicken und User-Details öffnen`, async () => {
+      await UserManagement.text_h2.click();
+      await page.getByRole('cell', { name: 'Max' }).click();
+    })
+
+    await test.step(`In den User-Details PW-Reset Dialog starten`, async () => {
+      await UserManagementDetail.text_h2.click();
+      await UserManagementDetail.button_pwChange.click();
+      await UserManagementDetail.text_warning_pwReset.click();
```

Comment:
Wieso wird der Text angeklickt?

Reply:
Damit wird sichergestellt, dass der Text vorhanden ist. Wenn der Anweder im Browser drauf klicken kann, ist dieser auch tatsächlich sichbar.

Reply:
Von der Logik sehe ich das ein, aber ich finde das semantisch etwas unverständlich. Ich fänd sowas wie .toBeVisible() aussagekräftiger.

Reply:
Die Playwright Tests sollen möglichst realistisch das echte Benutzerverhalten darstellen. Ein click() macht folgendes: (https://playwright.dev/docs/actionability)
- locator resolves to an exactly one element
- element is Visible
- element is Stable, as in not animating or completed animation
- element Receives Events, as in not obscured by other elements
- element is Enabled

Das ist semantisch nicht schön, aber ich würde sagen zuverlässiger, so zumindest meine Erfahrungen. Ich mache das nicht immer so, aber insgesamt müssen die Tests stabil sein.

Wir können es auch semantisch korrekt machen, und mal sehen ob es stabil ist.

Reply:
Okay, du hast da mehr Erfahrung. Können wir das irgendwo kurz festhalten? Vllt in der Readme oder in einem Code-Kommentar?

Reply:
Habe es jetzt mit expect und toBeVisible gemacht. Das sollte gut funktionieren und dann auch immer so gemacht werden wenn es darum geht, zu prüfen ob ein Textelement da ist. Ich hatte vorher nicht berücksichtigt, das expect hier entscheidend ist.